packages/integrations/vercel/src/lib/fs.ts

```diff
@@ -74,11 +75,13 @@ export async function copyFilesToFunction(
 
 		if (isSymlink) {
 			const realdest = fileURLToPath(new URL(nodePath.relative(commonAncestor, realpath), outDir));
-			await fs.symlink(
-				nodePath.relative(fileURLToPath(new URL('.', dest)), realdest),
-				dest,
-				isDir ? 'dir' : 'file'
-			);
+			const target = nodePath.relative(fileURLToPath(new URL('.', dest)), realdest);
+			// NOTE: when building function per route, dependencies are linked at the first run, then there's no need anymore to do that once more.
+			// So we check if the destination already exists. If it does, move on.
```

Comment:
But doesn't every function get its own `node_modules`?

Reply:
Yes, but inside a function we have symbolic links: 

<img width="381" alt="Screenshot 2023-09-01 at 13 16 08" src="https://github.com/withastro/astro/assets/602478/ac1cb91d-ab1f-4ec4-926b-c82bd97f3453">


This is `pnpm` doing its own thing I guess

Reply:
Full disclosure, it's still something a bit foggy to me, especially that part of the code. It's possible my comment is wrong